### PR TITLE
AWS OVS CI environment 

### PR DESCRIPTION
**Pull request by Ardiea**
## Description
Starter PR for Issue #1372

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (improves on existing behavior)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project. (Did you install and run the pre-commit hooks?)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Ardiea** [comment]
We should find a way to define a relationship between the Vault DB mount + the RDS instance that prevents the RDS instance from being removed prior to the vault mount being decomm'd and the vault leases being revoked. This is probably trivial to do. 

```
Diagnostics:
  vault:index:Mount (odlvideo-mount-point):
    error: deleting urn:pulumi:applications.odl_video_service.CI::ol-infrastructure-odl-video-service-env::ol:services:Vault:DatabaseBackend:postgresql$vault:index/mount:Mount::odlvideo-mount-point: 1 error occurred:
    	* error deleting from Vault: Error making API request.

    URL: DELETE https://vault-ci.odl.mit.edu/v1/sys/mounts/postgres-odl-video-service
    Code: 400. Errors:

    * failed to revoke "postgres-odl-video-service/creds/approle/n7VcGttirY72NyOyHSEHQVi8" (1 / 2965): failed to revoke entry: resp: (*logical.Response)(nil) err: failed to find entry for connection with name: "odlvideo"
```

**Comment by blarghmatey** [comment]
It should be as simple as passing `pulumi.ResourceOptions(depends_on=[<rds_component_resource_object>])`

**Comment by Ardiea** [comment]
AFAIK the OVS CI environment deployed in Heroku was never completed or functional so I don't think there is anything to migrate. That makes this a net new environment. 

Need to do the following:
- [ ] Double check stack configuration
- [ ] Deploy CI stack
- [ ] Verify the CI environment is up and working as expected
- [ ] Update deployment pipeline to incorporate CI environment
- [ ] Remove heroku application for OVS-CI

Optional items to populate CI env:
- [ ] Pull down RC database and install into CI? 
- [ ] Sync RC buckets into CI?

File: src/ol_concourse/lib/jobs/infrastructure.py [hunk 1]
```diff
@@ -139,11 +139,12 @@ def packer_jobs(  # noqa: PLR0913
     )
 
 
-def pulumi_jobs_chain(  # noqa: PLR0913, C901
+def pulumi_jobs_chain(  # noqa: PLR0912, PLR0913, C901
     pulumi_code: Resource,
     stack_names: list[str],
     project_name: str,
     project_source_path: Path,
+    enable_github_issue_resource: Optional[bool] = True,  # noqa: FBT002
```
Comment (blarghmatey) [inline src/ol_concourse/lib/jobs/infrastructure.py:147]:
```suggestion
    enable_github_issue_resource: bool = True,  # noqa: FBT002
```

**Review by blarghmatey — COMMENTED** [review]
In terms of separating the AMIs for CI and RC/Production you could add a tag in the Packer step. I think our eventual goal should be to marginally decouple the build/deploy of containers with the build/deploy of AMIs.

File: src/ol_concourse/pipelines/infrastructure/ovs/docker_packer_pulumi_pipeline.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ def build_ovs_pipeline() -> Pipeline:
     ovs_packer_code = git_repo(
         name=Identifier("ol-infrastructure-packer-build"),
         uri="https://github.com/mitodl/ol-infrastructure",
-        branch="main",
+        ranch="main",
```
Comment (blarghmatey) [inline src/ol_concourse/pipelines/infrastructure/ovs/docker_packer_pulumi_pipeline.py:57]:
```suggestion
        branch="main",
```

**Review by blarghmatey — APPROVED** [review]

**Pull request merged**